pkg/client/cli/cloud_messages.go

```diff
@@ -0,0 +1,157 @@
+package cli
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	empty "google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/spf13/cobra"
+
+	"github.com/datawire/dlib/dtime"
+
+	"github.com/telepresenceio/telepresence/rpc/v2/systema"
+	"github.com/telepresenceio/telepresence/v2/pkg/client"
+	"github.com/telepresenceio/telepresence/v2/pkg/client/cache"
+	"github.com/telepresenceio/telepresence/v2/pkg/client/cli/cliutil"
+)
+
+const messagesCacheFilename = "cloud-messages.json"
+
+type cloudMessageCache struct {
+	NextCheck         time.Time       `json:"next_check"`
+	Intercept         string          `json:"intercept"`
+	MessagesDelivered map[string]bool `json:"messagess_delivered"`
+}
+
+// newCloudMessageCache returns a new CloudMessageCache, initialized from the users' if it exists
+func newCloudMessageCache(ctx context.Context) (*cloudMessageCache, error) {
+	cmc := &cloudMessageCache{}
+
+	if err := cache.LoadFromUserCache(ctx, cmc, messagesCacheFilename); err != nil {
+		if !os.IsNotExist(err) {
+			return nil, err
+		}
+		cmc.NextCheck = time.Time{}
+		cmc.MessagesDelivered = make(map[string]bool)
+		_ = cache.SaveToUserCache(ctx, cmc, messagesCacheFilename)
+	}
+	return cmc, nil
+}
+
+// getCloudMessages communicates with Ambassador Cloud and stores those messages
+// in the cloudMessageCache.
+// For now this function will just return a fake result from the cloud
+// this will be replaced with the code to talk to Ambassador Cloud
+// via GetUnauthenticatedCommandMessages in a later commit
```

Comment:
this comment seems like it's out of date

Reply:
🤦